app/models/set_member_subject.rb

```diff
@@ -53,7 +54,7 @@ def self.unseen_for_user_by_workflow(user, workflow)
   end
 
   def retire_workflow(workflow)
-    count = subject_workflow_counts.find_or_create_by!(workflow_id: workflow.id)
+    count = SubjectWorkflowCount.find_or_create_by!(workflow_id: workflow.id, subject_id: subject_id)
     count.retire!
```

Comment:
Should we kick off the dequeue for retired subjects here similar to what i've proposed in #1339?